### PR TITLE
Failure to sync no longer logs tracebacks, also reports a more helpful message

### PR DESCRIPTION
**Pull request by mhrivnak**
This depends on changes in pulp that better-handle PulpCodedException reporting.

fixes #702
refs #652

https://pulp.plan.io/issues/652
https://pulp.plan.io/issues/702

https://github.com/pulp/pulp/pull/1772

### CODE REVIEW

File: plugins/test/unit/plugins/test_registry.py [hunk 1]
```diff
@@ -1,15 +1,21 @@
 from cStringIO import StringIO
 import json
 import os
+import shutil
 import tempfile
-import unittest
+
+try:
+    import unittest2 as unittest
```
Comment (beav) [inline plugins/test/unit/plugins/test_registry.py]: Can we assume that `unittest2` is always available?

Comment (mhrivnak) [inline plugins/test/unit/plugins/test_registry.py]: I addressed that in this PR: https://github.com/pulp/pulp/pull/1772

``unittest2`` will be required on EL6, but not for EL7 for F*. I added it to the spec file for pulp-devel on el6, and to the puppet class used by our test automation.

Comment (beav) [inline plugins/test/unit/plugins/test_registry.py]: ah, cool

**Comment by beav** [comment]
LGTM :sunglasses:

**Comment by mhrivnak** [comment]
I bumped the required version of pulp to 2.7, which I forgot to do initially.

**Pull request merged**